flowzone.yml

```diff
@@ -1180,34 +1180,45 @@ jobs:
           echo "json=$(echo $response | jq -c .)" >> $GITHUB_OUTPUT
 
       # https://docs.github.com/en/rest/git/refs?apiVersion=2022-11-28#update-a-reference
-      - name: Update branch reference
+      - name: Update git references
         if: github.event.pull_request.merged == true && steps.create_commit.outputs.sha != ''
+        # some tags may not initially exist (e.g. production)
+        continue-on-error: true
         env:
           GH_TOKEN: ${{ steps.gh_app_token.outputs.token || secrets.FLOWZONE_TOKEN }}
         run: |
-          gh api \
-            -X PATCH \
-            -H "Accept: application/vnd.github+json" \
-            -H "X-GitHub-Api-Version: 2022-11-28" \
-            /repos/$GH_REPO/git/refs/heads/${{ github.base_ref }} \
-            -f sha='${{ steps.create_commit.outputs.sha }}' \
-            -F force=true \
-            --include
+          for ref in \
+            '/repos/$GH_REPO/git/refs/heads/${{ github.base_ref }}' \
+            '/repos/$GH_REPO/git/refs/tags/production'; do
```

Comment:
The production tag seems arbitrary, especially without a commit message or comment to explain where it is used.

Maybe we should create an env or input for floating tags that need to be updated on merge rather than hardcoding something that is only used for a few internal repos.

Then we wouldn't need to ignore failures if the floating tag is provided by the calling repo.

Reply:
`balena-deploy-request`, which is used to generate release notes, requires this tag. It should probably be set as a default to `production`, with a way to override it (though I am struggling to see a use case for the later).

Ultimately we want this in as a prerequisite to Flowzone versioning to add a PR comment to the PR for the approver to adjust (notable changes) prior to merging.

Reply:
Yup, I understand the use case but no one in the future looking at this code will know why it is there and why it fails on most repos except some balena backend components.

It just seems like something that should be opt-in.

Reply:
OK, how do you propose this opt-in to work? Another input (e.g. `generate_release_notes`)?

Reply:
An input like floating_tags might do it with a good enough description.

Reply:
`balena-deploy-request` requires minimum of two tags in the format `production-$(date +%Y-%m-%d-%H%M%S)`, so the input would have to be something like `floating_tag_prefix` or just a boolean `floating_tags`, or some sort of weird template format, which we would then have to parse and substitute env vars, etc.

I suggest a simple boolean input called `release_notes`, which will toggle the entire workflow (production tags and PR comment).

Reply:
@klutchell thoughts, prayers?

Reply:
> I suggest a simple boolean input called `release_notes`, which will toggle the entire workflow (production tags and PR comment).

This is probably the best approach, go for it.